backend/main.py

```
@@ -48,21 +99,36 @@ async def root():
 async def get_projects():
     """Get all projects"""
     try:
-        return file_service.load_projects()
+        logger.info("Loading projects")
+        projects = file_service.load_projects()
+        logger.info(f"Loaded {len(projects)} projects")
+        return projects
     except Exception as e:
-        raise HTTPException(status_code=500, detail=str(e))
+        logger.error(f"Error loading projects: {e}")
+        raise HTTPException(status_code=500, detail=f"Failed to load projects: {str(e)}")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)